File: desktop/src/renderer/models/llm/LLMStatusResponse.ts
import type { InstalledModel } from "./InstalledModel";

/** Réponse de l'endpoint statut LLM. */
export interface LLMStatusResponse {
  available: boolean;
  message: string;
  selected_model_id?: string | null;
  installed_models?: InstalledModel[];
}

File: backend/core/models/llm/llmStatusResponse.py
from dataclasses import dataclass, field

from backend.core.models.llm.installedModelResponse import InstalledModelResponse


@dataclass
class LLMStatusResponse:
    available: bool
    message: str
    resolved_model_path: str | None = None
    resolved_model_exists: bool | None = None
    resources_path: str | None = None
    selected_model_id: str | None = None
    installed_models: list[InstalledModelResponse] = field(default_factory=list)
